### ISSUE TITLE
[BUG] Long log lines are not readable - word wrap mode?

### ISSUE THREAD

**Issue by feliperaul** [mrjackwills/oxker#54]
When looking at the logs, oxker displays them one line at a time without any word wrap; is there a way to make the lines wrap so that I can read them?

**Comment by mrjackwills** [comment]
Yes I completely agree, [oxker](https://www.github.com/mrjackwills/oxker) uses the [ratatui](https://github.com/ratatui/ratatui) library to create its interface. There is an open [RFC](https://github.com/ratatui/ratatui/issues/293) about text wrapping, but as far as I can tell there isn't an easy solution at the moment.

Although, there is na open [issue](https://github.com/ratatui/ratatui/issues/128) with some solutions, so I can have a look how they would work.

**Comment by burakcank** [comment]
Is there any progress regarding to this? It's interesting that such a fundamental feature is currently not working as expected.

**Comment by mrjackwills** [comment]
The issues are still open in the [ratatui](https://github.com/ratatui/ratatui/issues) repo.

I have a brief look at it, but couldn't get it to work as expected, doing so would require substantial modification to `oxker`. I do agree that that feature is needed, but I'd need to commit a decent amount of time to get it to an acceptable state.

**Comment by mrjackwills** [comment]
@marlop352 That's a pretty decent idea, can use `←` and `→` to move left and right along the logs line, can even save the position of the offset for each container, as in if you scroll right 50 characters for `container_a`, then switch to read `container_b`, and then back to `container_a`, the 50 offset would still be in place. Should indicate to the user that this is the case though

**Comment by mrjackwills** [comment]
@marlop352 would you be able to test a build with this feature in it? What platform does it need to be built for, or would you rather build from source?

**Comment by marlop352** [comment]
@mrjackwills I would need a binary for x86_64.

**Comment by marlop352** [comment]
It seems I got here too late to test it 😅
@mrjackwills  do ping me when you have another build. It may take me a while but I will test it.

**Comment by mrjackwills** [comment]
@marlop352 better late than never, this build should *hopefully* work.

Thanks, there should be a binary attached to this comment, the controls are ← and → to scroll the logs horizontally. This can be adjusted in the keymap with the new log_scroll_forward and log_scroll_back keys

1. It should only work when the logs panel is selected.
2. A counter should appear in the bottom left of the log panel, displaying how much horizontal scroll is in use and available.
3. It should be impossible to scroll off the screen.

Please let me know how it goes.

As far as I can tell, across multiple systems, it works as expected. In which case a new release should be coming soon

[oxker_linux_x86_scroll.zip](https://github.com/user-attachments/files/21780746/oxker_linux_x86_scroll.zip)

**Comment by marlop352** [comment]
Finally had some time to test this.

Not sure if that was because of windows terminal but sometimes it leaves artifacts:

<img width="1045" height="1003" alt="Image" src="https://github.com/user-attachments/assets/5a131e69-06fb-466b-a0fc-4086ef39b84f" />

Some other comments:
- it would be better if the counter was always visible in the log panel if horizontal scroll is available
- I would love if ctrl + ← and ctrl + → went by increments of 10 to speedup logs navigation 😅, scrolling this can take some time at normal speed: <img width="84" height="40" alt="Image" src="https://github.com/user-attachments/assets/a9dd207f-4663-4d26-9ea8-43ee7df4bcd0" />

other than that it worked pretty well.

**Comment by mrjackwills** [comment]
Thanks, I've not seen those artifacts, but the latest build has a new feature, to completely clear the screen and redraw, by default i uses the `f` key. Again there should be an attachment of that build to this comment.

As for always showing the counter, that probably makes sense.
`ctrl` + `←`/ `→` also would be a nice feature, I think there's another issue open for something similar, so that should also be implemented.

[oxker_x86_linux.zip](https://github.com/user-attachments/files/21817392/oxker_x86_linux.zip)

**Comment by mrjackwills** [comment]
Yeah I can switch `shift` to `ctrl`, although in the release it'll be configurable by using a `config.toml`, there's also another [issue](https://github.com/mrjackwills/oxker/issues/69) that suggest use of the`shift` key, so that would make it easier.

As for artifacts, again, I see them inconsistently, I can't fathom where they are being created. I know I have rebooted the remote machine, and then using `oxker` on it (via ssh), the artifacts were no longer present, but that isn't really a solution, nor cause.
I've also seen issues when a log file contains chars that could be interpreted as control characters, such as the line

`[2025-08-20 13:01:00.698][readsb] Database update first part took 0.675 seconds!\n`.

You can check for these using the raw mode, via `-r` cli argument - or `config.toml` file. Although, in my case, this only caused issues when `oxker` was using it's colourise mode (via `-c` cli argument, or `config.toml` file).

Else probably expect a new release with this feature soon.

**Comment by mrjackwills** [comment]
Closing for now, but maybe re-open in anything changes in term of inbuilt features of the [ratatui](https://github.com/ratatui/ratatui) library
